app/services/idv/financials_step.rb

```diff
@@ -0,0 +1,32 @@
+module Idv
+  class FinancialsStep < Step
+    def complete?
+      idv_form.finance_type && idv_session.financials_confirmation.try(:success?) ? true : false
+    end
+
+    private
+
+    def vendor_validate
+      result = vendor_validator.validate
+      idv_session.params = idv_form.idv_params if complete?
```

Comment:
This does not belong here IMO. The Vendor Validator class only has one job to do: to validate the data sent to it. It should not be concerned with mutating anything. This session manipulation belongs in the only public method that this class should have: the one that performs all the tasks and returns a result hash. See https://github.com/18F/identity-idp/pull/947 for what I'm proposing.